packages/react-components/react-tree/src/components/TreeItem/useTreeItem.ts

```diff
@@ -133,103 +123,69 @@ export function useTreeItem_unstable<Value = string>(
       case ArrowLeft:
         return handleArrowLeft(event);
       case End:
-        return requestNavigation({ event, value, type: treeDataTypes.end, target: event.currentTarget });
       case Home:
-        return requestNavigation({ event, value, type: treeDataTypes.home, target: event.currentTarget });
       case ArrowUp:
-        return requestNavigation({ event, value, type: treeDataTypes.arrowUp, target: event.currentTarget });
       case ArrowDown:
-        return requestNavigation({ event, value, type: treeDataTypes.arrowDown, target: event.currentTarget });
+        return requestNavigation({ event, value, type: treeDataTypes[event.key], target: event.currentTarget });
     }
     const isTypeAheadCharacter =
       event.key.length === 1 && event.key.match(/\w/) && !event.altKey && !event.ctrlKey && !event.metaKey;
     if (isTypeAheadCharacter) {
-      return requestNavigation({ event, value, target: event.currentTarget, type: treeDataTypes.typeAhead });
+      return requestNavigation({ event, value, target: event.currentTarget, type: treeDataTypes.TypeAhead });
     }
   });
 
-  const [isActionsVisible, setActionsVisible] = React.useState(false);
-  const showActions = useEventCallback((event: React.SyntheticEvent) => {
-    const isEventFromSubtree = subtreeRef.current && elementContains(subtreeRef.current, event.target as Node);
-    if (!isEventFromSubtree) {
+  const handleActionsVisible = useEventCallback((event: React.FocusEvent) => {
```

Comment:
the naming of these functions is really confusing, in `handleActionsInvisible` you can also set the actions to be visible.

in `handleActionsVisible` you don't always set actions to be visible